api/openapi_server/controllers/host_controller.py

```diff
@@ -1,40 +1,17 @@
-import traceback
-import connexion
+from flask import Response
 
-from openapi_server.models.host import Host
-from openapi_server.models import database as db
-from sqlalchemy.orm import Session
-from sqlalchemy.ext.serializer import loads, dumps
-
-db_engine = db.DataAccessLayer.get_engine()
+from openapi_server.models.database import hostSchema, hostsSchema, Host, DataAccessLayer
 
 # create host in database
-def create_host():
-    if connexion.request.is_json:
-        try:
-            host = Host.from_dict(connexion.request.get_json()).to_dict()
-        except ValueError:
-            return traceback.format_exc(ValueError), 400
-    
-    with Session(db_engine) as session:
-      row = db.Host(
-        name=host["name"]
-      )
-
-      session.add(row)
-      session.commit()
-
-      host["id"] = row.id
-      return Host.from_dict(host), 201
-
-def get_hosts():
-    resp = []
-
-    with Session(db_engine) as session:
-        table = session.query(db.Host)
-
-        for row in table:
-            host = Host(id=row.id, name=row.name).to_dict()
-            resp.append(Host.from_dict(host))
-
-    return resp, 200
+def create_host(body: dict) -> Response:
+    with DataAccessLayer.session() as session:
+        new_host = Host(name = body["name"])
+        session.add(new_host)
+        session.commit()
+        _ = new_host.id
```

Comment:
Thanks for calling this out! Short answer is yes, accessing the id does force `SQLAlchemy` to refresh the instance so that it is synchronized with the newly added/updated database row. You can verify this by commenting out the line and running `pytest` to see the failures.

This is a hacky way of forcing the refresh, so I took a dive into the `SQLAlchemy` docs to see if there is a better solution.

There are three things to be aware of here:

1. Returning the `SQLAlchemy` python instance like this detaches the instance from the session. This means the instance has no attachment to the database, unless it is attached to a new session. Any attempt to load an uninitialized property will result in a runtime exception, so you need to be careful to only detach fully refreshed instances.
2. Calling `commit` on a `SQLAlchemy` python instance will mark that instance as [expired](https://docs.sqlalchemy.org/en/14/orm/session_basics.html#committing). This means that newly committed instances are out-of-date with the database, and they need to be refreshed in order to properly reflect all the properties (such as the id in our case). This refresh is normally done behind-the-scenes, but we need to force it since we are detaching it from the session.
3. Calls to `flush` will not update the SQLAlchemy python instance. Flushing transfers in-memory session changes to the database, without altering the SQLAlchemy python instance at all. 

So there are at least two options to ensure the id is available to the endpoint:

1. Do not detach the instance from the session. If the endpoint passes in a session to the repository method, then the repo method will still return an expired instance. This, however, won't be a problem because the expired instance will be attached to a session and therefore will be able to update the properties whenever they are needed. 
2. Instead of calling `_ = newProvider.id` to force the refresh of the id field, we can call `session.refresh(newProvider)` to guarantee that all properties are refreshed before detaching it from the instance. 

Option 1 would require passing a session instance into the repository methods. I think this would make using them much more complicated, so I'll use Option 2.

Reply:
Fix the refresh hack in [1d8a319](https://github.com/hackforla/HomeUniteUs/pull/575/commits/1d8a319d4d549ac1424390e5c468d8689aba3f97)